R/join.R

```diff
@@ -0,0 +1,91 @@
+#' @export
+semi_join.tbl_svy <- function(
+  x,
+  y,
+  by = NULL,
+  copy = FALSE,
+  ...,
+  na_matches = c("na", "never")
+) {
+
+  if (inherits(y, "tbl_svy")) {
+    y <- y$variables
+  }
+
+  x <- mutate(x, `___row_number` = dplyr::row_number())
+
+  filtered_vars <- semi_join(x = x$variables,
+                             y = y,
+                             by = by,
+                             copy = copy,
+                             na_matches = na_matches,
+                             ...)
+
+  x <- mutate(x, `___retained` = `___row_number` %in% filtered_vars[['___row_number']])
```

Comment:
Can you skip the `___retained` variable and just filter on the expression? (If not, I thnk you need to drop `___row_number`, don't you?)

Reply:
Yeah, that's a good idea. I didn't de-select `___row_number` because it was actually removed internally in the call to `filter()`. But just to be careful, in f240676 added a conditional select to remove that column if it still exists after using `filter()`.